### PR TITLE
[DOC] use new surface API in plot_3d_map_to_surface_projection

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #4568

Changes proposed in this pull request:

- part of moving surface API to stable codebase
- replace plot_3d_map_to_surface_projection with example taken from experimental

### CODE REVIEW

**Comment by bthirion** [comment]
Let us know when this is ready for review.

**Review by man-shu — APPROVED** [review]
LGTM!

File: examples/01_plotting/plot_3d_map_to_surface_projection.py [hunk 1]
```diff
@@ -136,92 +146,93 @@
...
-        legend=True,
-        colors=["g", "k"],
-    )
-    plotting.show()
-elif engine == "plotly":
-    figure.add_contours(
-        roi_map=parcellation,
-        levels=regions_indices,
-        labels=labels,
-        lines=[{"width": 5}],
```
Comment (Remi-Gau) [inline examples/01_plotting/plot_3d_map_to_surface_projection.py:190]: need to check why this was removed

File: examples/01_plotting/plot_3d_map_to_surface_projection.py [hunk 2]
```diff
@@ -136,161 +146,182 @@
...
         levels=regions_indices,
         figure=figure,
         legend=True,
         colors=["g", "k"],
     )
-    plotting.show()
+    show()
 elif engine == "plotly":
     figure.add_contours(
-        roi_map=parcellation,
```
Comment (Remi-Gau) [inline examples/01_plotting/plot_3d_map_to_surface_projection.py:187]: This is one function we have not adapted yet to accept SurfaceImage yet
Will open an issue to fix this.

**Review by bthirion — APPROVED** [review]
Happy to remove code !

**Comment by Remi-Gau** [comment]
doc build failure is unrelated: happens on main is should be fixed by #4694

**Pull request merged**